src/content/app/entity-viewer/variant-view/variant-view-sidebar/overview/MainAccordion.tsx

```diff
@@ -106,39 +122,79 @@ const MainAccordion = (props: EntityViewerVariantDefaultQueryResult) => {
   );
 };
 
-type AlleleProps = {
-  alleles: {
-    allele_sequence: string;
-    reference_sequence: string;
-    allele_type: {
-      value: string;
-    };
-  }[];
+type AllelesProps = {
+  genomeId: string;
+  variantId: string;
+  activeAlleleId: string | null;
+  alleles: VariantDetails['alleles'];
 };
-const Alleles = (props: AlleleProps) => {
-  const { referenceAllele, alternativeAlleles } = getReferenceAndAltAlleles(
-    props.alleles
-  );
+
+const Alleles = (props: AllelesProps) => {
+  const { genomeId, variantId, alleles, activeAlleleId } = props;
+  const { referenceAllele, alternativeAlleles } =
+    getReferenceAndAltAlleles(alleles);
 
   return (
     <div>
-      <div className={styles.row}>
-        <div className={styles.label}>Ref</div>
-        <div className={styles.value}>
-          {referenceAllele?.reference_sequence}
+      {referenceAllele && (
+        <div className={styles.row}>
+          <div className={styles.label}>Ref</div>
+          <div className={styles.value}>
+            <Allele
+              genomeId={genomeId}
+              variantId={variantId}
+              allele={referenceAllele}
+              activeAlleleId={activeAlleleId}
+            />
+          </div>
         </div>
-      </div>
+      )}
       <div className={styles.row}>
         <div className={styles.label}>Alt</div>
-        <div className={styles.value}>
-          {alternativeAlleles.map((allele, index) => (
-            <div key={index} className={styles.alleleSequence}>
-              {allele.allele_sequence}
-            </div>
+        <div className={styles.altAlleles}>
+          {alternativeAlleles.map((allele) => (
+            <Allele
+              key={allele.urlId}
+              genomeId={genomeId}
+              variantId={variantId}
+              allele={allele}
+              activeAlleleId={activeAlleleId}
+            />
           ))}
         </div>
       </div>
     </div>
   );
 };
+
+const Allele = (props: {
+  genomeId: string;
+  variantId: string;
+  allele: VariantDetails['alleles'][number];
+  activeAlleleId: string | null;
+}) => {
+  const { genomeId, variantId, allele, activeAlleleId } = props;
+  const formattedSequence = formatAlleleSequence(allele.allele_sequence);
+
+  if (allele.urlId === activeAlleleId) {
+    return <span>{formattedSequence}</span>;
+  } else {
+    const url = urlFor.entityViewerVariant({
+      genomeId,
+      variantId,
+      alleleId: allele.urlId
+    });
+
+    return <Link to={url}>{formattedSequence}</Link>;
+  }
+};
+
+const formatAlleleSequence = (sequence: string) => {
+  const maxCharacters = 18;
```

Comment:
Max as per Andrea is 18 characters including ellipsis

Reply:
Yes, line 196 is handling that for ellipsis.

Reply:
Not a big problem, but trying understand the code, isn't it 18 + 3 dots = 21 character?

Reply:
It isn't three individual dot characters; it is one character that has three dots in it 😄  

https://www.compart.com/en/unicode/U+2026